device/alpaka/src/clusterization/clusterization_algorithm.cpp

```diff
@@ -74,6 +95,23 @@ clusterization_algorithm::output_type clusterization_algorithm::operator()(
     auto devAcc = ::alpaka::getDevByIdx(::alpaka::Platform<Acc>{}, 0u);
     auto queue = Queue{devAcc};
 
+    // Setup the mutex, if it is not already setup.
+    while (!m_mutex_is_setup) {
+        if (m_setup_mutex.try_lock()) {
+            if (!m_mutex_is_setup) {
+                m_mutex_is_setup = true;
+
+                auto workDiv = makeWorkDiv<Acc>(1, 1);
+
+                ::alpaka::exec<Acc>(queue, workDiv, ZeroMutexKernel{},
+                                    m_backup_mutex.get());
+                ::alpaka::wait(queue);
```

Comment:
Remind me, why is a memset operation not an option here instead of running a kernel with a single thread? :confused:

Reply:
I think Alpaka's memset functionality only works with its own built-in buffers accessors; vecmem has the same restriction, e.g. it only works with buffers, etc. That left me with the option of either using preprocessor macros to differentiate CPU/CUDA/HIP/etc. compilation, or to launch a kernel to simply set the flag, which turned out to be a lot easier.